File: Classes/Premade/MASPreferences/PrefSetter.m
//
//  PrefSetter.m
//  BeFit
//
//  Created by Jon Brown on 1/5/14.
//
//

#import "PrefSetter.h"

@implementation PrefSetter

-(void)awakeFromNib
{
    [prefslider setTarget:self]; // assume the handler is [self sliderDidMove:]
    [prefslider setAction:@selector(sliderDidMove:)];

    NSNumberFormatter *slidervalueFormatter = [[NSNumberFormatter alloc] init];
    [slidervalueFormatter setFormat: @"#,###;0;(#,##0)"];
    [[sliderValueLabel cell] setFormatter:slidervalueFormatter];

    
    NSUserDefaults* defaults = [NSUserDefaults standardUserDefaults];
    NSString*  goalPrefs = [defaults objectForKey:@"goal-name"];
    
    // Read default prefrences for member name
    if (goalPrefs == (id)[NSNull null] || goalPrefs.length == 0 ) {
        
        [prefslider setStringValue:@"2000"];
        [sliderValueLabel setStringValue:@"2000"];
        
    } else {
        
        [prefslider setStringValue:goalPrefs];
        [sliderValueLabel setStringValue:goalPrefs];
    }
    
}

- (IBAction)sliderDidMove:(id)sender {
    
    NSSlider *slider = sender;
    double textValue = [slider doubleValue];
    [sliderValueLabel setDoubleValue:textValue];
    
    NSUserDefaults* defaults = [NSUserDefaults standardUserDefaults];
    NSString *sliderPreferenceValue = [prefslider stringValue];
    
    if (sender == prefslider)
    {
        [defaults setObject:sliderPreferenceValue forKey:@"goal-name"];
        
    }
    
}

@end

File: GUI Objects/PrefSetter.swift
//
//  PrefSetter.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/8/19.
//

import Foundation
import Cocoa

class PrefSetter: NSObject, NSApplicationDelegate {
    @IBOutlet var prefslider: NSSlider!
    @IBOutlet var sliderValueLabel: NSTextField!
    
    var textValue: String?
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        prefslider.target = self // assume the handler is [self sliderDidMove:]
        prefslider.action = #selector(sliderDidMove)

        let slidervalueFormatter = NumberFormatter()
        slidervalueFormatter.format = "#,###;0;(#,##0)"
        sliderValueLabel.cell?.formatter = slidervalueFormatter
        
        let goalPrefs = UserDefaults.standard.string(forKey: "goal-name")
        
        // Read default prefrences for member name
        if goalPrefs?.isEmpty ?? true {
            prefslider.stringValue = "2000"
            sliderValueLabel.stringValue = "2000"
        } else {
            prefslider.stringValue = goalPrefs ?? ""
            sliderValueLabel.stringValue = goalPrefs ?? ""
        }
    }
    
    @objc private func sliderDidMove(_ sender: NSSlider) {
        let textValue = sender.doubleValue
        sliderValueLabel.doubleValue = textValue
        
        let sliderPreferenceValue = prefslider.stringValue
        
        if sender == prefslider {
            UserDefaults.standard.set(sliderPreferenceValue, forKey: "goal-name")
        }
    }
}
